src/router/router.ts

```diff
@@ -196,6 +208,31 @@ export class DanetHTTPRouter {
 		};
 	}
 
+	private handleSSE(executionContext: ExecutionContext, response: EventTarget) {
+		return streamSSE(executionContext, async (stream: SSEStreamingApi) => {
+			let canContinue = true;
+			(response as unknown as EventTarget).addEventListener(
+				'message',
+				async (event) => {
+					const { detail: payload } = event as SSEEvent;
+					await stream.writeSSE({
+						data: payload.data,
+						event: payload.event,
+						id: payload.id,
+						retry: payload.retry,
+					});
+					if (payload.event === 'close') {
+						canContinue = false;
+					}
```

Comment:
If `event === 'close'`, this function will first call `writeSSE` with the close event and then it'll close the stream.

Is that expected, of should we close the stream without sending the close event?

Reply:
I think writing the close event is more flexible, as it gives the user the ability to send additional information in `data`

Reply:
Makes sense, sounds good to me! 😁